### PR TITLE
copy issmirnov/qmk-keebs Feb 20, 2021 into QMK mainline

### PR DESCRIPTION
**Pull request by issmirnov**
Copy in @issmirnov layouts from https://github.com/issmirnov/qmk-keebs

I have been deeply helped by sample layouts of others, so I'd like to update mine in turn so that they may inspire others.

## Description

Misc updates to keymaps, RGB underglow, and space optimizations.

## Types of Changes

- [X] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).

### CODE REVIEW

File: keyboards/keebio/levinson/keymaps/issmirnov/config.h [hunk 1]
```diff
@@ -9,10 +13,22 @@
    // Enable animations. +5500 bytes
    #define RGBLIGHT_ANIMATIONS
 
+   // remove unused animations
```
Comment (drashna) [inline keyboards/keebio/levinson/keymaps/issmirnov/config.h:16]: might be better to undef `RGBLIGHT_ANIMATIONS` and only explicitly define the ones you want.

Comment (issmirnov) [reply keyboards/keebio/levinson/keymaps/issmirnov/config.h]: Done, thanks!

File: keyboards/keebio/levinson/keymaps/issmirnov/config.h [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
 // DO NOT enable USE_IDC - board will not respond.
 #define USE_SERIAL
 //#define USE_I2C
 
 #ifdef RGBLIGHT_ENABLE
-   // Enable animations. +5500 bytes
-   #define RGBLIGHT_ANIMATIONS
 
+    // Disable all animations
+    // #undef RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/keebio/levinson/keymaps/issmirnov/config.h:15]:
```suggestion
    #undef RGBLIGHT_ANIMATIONS
```

**Review by drashna — COMMENTED** [review]
It also looks like you have changes to the submodules.  You may want/need to run `make git-submodule` and add a commit to fix this issue.

**Comment by issmirnov** [comment]
@drashna Can I get some help with the submodules? I ran `make git-submodule` but it didn't generate a diff. `git submodule init` and `git submodule update`
didn't seem to help either.

File: users/issmirnov/rules.mk [hunk 3]
```diff
@@ -4,6 +4,7 @@ SRC += issmirnov.c
 
 # https://www.reddit.com/r/olkb/comments/bmpgjm/programming_help/
 # Should shave 2000 bytes
+# Formerly known as LINK_TIME_OPTIMIZATION_ENABLE
```
Comment (fauxpark) [inline users/issmirnov/rules.mk:7]:
```suggestion
```

File: users/issmirnov/rules.mk [hunk 4]
```diff
@@ -14,6 +15,7 @@ CONSOLE_ENABLE = yes # +400 bytes (hid_listen support)
 COMBO_ENABLE = yes
 
 # This allows the keyboard to tell the host OS that up to 248 keys are held down at once
+# NKRO_ENABLE = yes # note: also needs FORCE_NKRO in config.h
```
Comment (fauxpark) [inline users/issmirnov/rules.mk:18]:
```suggestion
```

File: layouts/community/ergodox/issmirnov/rules.mk [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# https://github.com/qmk/qmk_firmware/blob/master/docs/getting_started_make_guide.md
+ifndef QUANTUM_DIR
+    include ../../../../Makefile
+endif
+
+# Enable ORYX options
+ORYX_ENABLE = yes
+WEBUSB_ENABLE = yes
+
```
Comment (fauxpark) [inline layouts/community/ergodox/issmirnov/rules.mk:9]:
```suggestion
# Enable ORYX options
ORYX_ENABLE = yes
WEBUSB_ENABLE = yes
```

File: keyboards/keebio/levinson/keymaps/issmirnov/ascii-keymap.txt [hunk 6]
```diff
@@ -0,0 +1,55 @@
+                                   Qwerty
```
Comment (fauxpark) [inline keyboards/keebio/levinson/keymaps/issmirnov/ascii-keymap.txt:1]: Please remove these text files, or put them into the readme.

File: keyboards/keebio/levinson/keymaps/issmirnov/push-left.sh [hunk 7]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (fauxpark) [inline keyboards/keebio/levinson/keymaps/issmirnov/push-left.sh:1]: These scripts should really go as well. Use the CLI instead, you can run it from anywhere too.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keebio/levinson/keymaps/issmirnov/push-left.sh (hunk 7 above)
Comment (issmirnov) [reply keyboards/keebio/levinson/keymaps/issmirnov/push-left.sh]: Done.

File: keyboards/keebio/levinson/keymaps/issmirnov/ascii-keymap.txt (hunk 6 above)
Comment (issmirnov) [reply keyboards/keebio/levinson/keymaps/issmirnov/ascii-keymap.txt]: Done.

**Review by issmirnov — COMMENTED** [review]
Updated all files.

**Comment by drashna** [comment]
`make git-submodule` doesn't help out here?

**Comment by issmirnov** [comment]
Unfortunately not. Any thoughts? I could copy the files manually into a new branch, but running `make git-submodule` is what got me into this mess in the first place.

Screen Shot 2021-04-05 at 3 43 06 PM

**Comment by drashna** [comment]
Those commit checkouts look to be correct, and the status is good at the end there. 

Try running: 
```
git commit --allow-empty -m "Fix git submodules"
```

**Pull request closed without merging**